lightning/src/ln/shutdown_tests.rs

```diff
@@ -726,3 +726,103 @@ fn test_invalid_shutdown_script() {
+	assert!(node_0_closing_signed.fee_satoshis <= 500);
+
+	if timeout_step != 0 {
+		nodes[1].node.handle_closing_signed(&nodes[0].node.get_our_node_id(), &node_0_closing_signed);
+		// At this point nodes[1] should send back a warning message indicating it disagrees with the
+		// given channel-closing fee. Currently we do not implement warning messages so instead we
+		// remain silent here.
+		assert!(nodes[1].node.get_and_clear_pending_msg_events().is_empty());
+
+		// Now deliver a mutated closing_signed indicating a higher acceptable fee range, which
+		// nodes[1] should happily accept and respond to.
+		node_0_closing_signed.fee_range.as_mut().unwrap().max_fee_satoshis *= 10;
+		{
+			let mut lock;
+			get_channel_ref!(nodes[0], lock, chan_id).closing_fee_limits.as_mut().unwrap().1 *= 10;
+		}
+		nodes[1].node.handle_closing_signed(&nodes[0].node.get_our_node_id(), &node_0_closing_signed);
+		let node_1_closing_signed = get_event_msg!(nodes[1], MessageSendEvent::SendClosingSigned, nodes[0].node.get_our_node_id());
+		nodes[0].node.handle_closing_signed(&nodes[1].node.get_our_node_id(), &node_1_closing_signed);
+		let node_0_2nd_closing_signed = get_closing_signed_broadcast!(nodes[0].node, nodes[1].node.get_our_node_id());
+		if timeout_step > 1 {
+			nodes[1].node.handle_closing_signed(&nodes[0].node.get_our_node_id(), &node_0_2nd_closing_signed.1.unwrap());
+		}
+	}
+
+	if timeout_step <= 1 {
+		assert!(nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap().is_empty());
+	} else {
+		assert_eq!(nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap().len(), 1);
+	}
+
+	nodes[1].node.timer_tick_occurred();
+	nodes[1].node.timer_tick_occurred();
+
+	let txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
+	assert_eq!(txn.len(), 1);
+	assert_eq!(txn[0].output.len(), 2);
+
+	// If we timed out we should have a P2WPKH and P2WSH output, otherwise both should be P2WPKH.
+	if timeout_step <= 1 {
+		assert!((txn[0].output[0].script_pubkey.len() == 20+1+1 &&
+		         txn[0].output[1].script_pubkey.len() == 32+1+1) ||
+		        (txn[0].output[1].script_pubkey.len() == 20+1+1 &&
+		         txn[0].output[0].script_pubkey.len() == 32+1+1));
+		check_closed_broadcast!(nodes[1], true);
+		check_added_monitors!(nodes[1], 1);
+	} else {
+		assert_eq!(txn[0].output[0].script_pubkey.len(), 20+1+1);
+		assert_eq!(txn[0].output[1].script_pubkey.len(), 20+1+1);
+
+		let events = nodes[1].node.get_and_clear_pending_msg_events();
+		assert_eq!(events.len(), 1);
+		match events[0] {
+			MessageSendEvent::BroadcastChannelUpdate { ref msg } => {
+				assert_eq!(msg.contents.flags & 2, 2);
+			},
+			_ => panic!("Unexpected event"),
+		}
+	}
+}
+
+#[test]
+fn test_closing_signed_reinit_timeout() {
+	do_test_closing_signed_reinit_timeout(0);
+	do_test_closing_signed_reinit_timeout(1);
+	do_test_closing_signed_reinit_timeout(2);
```

Comment:
Isn't super clear what `timeout_step` is meant to represent. It seems like `2` means it didn't timeout, ending in a cooperative close? But the others do not. Maybe an enum would clarify.